keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -0,0 +1,154 @@
+import dataclasses
+
+import pydantic
+from splunklib.client import connect
+
+from keep.api.models.alert import AlertDto, AlertSeverity
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig, ProviderScope
+from keep.providers.providers_factory import ProvidersFactory
+
+
+@pydantic.dataclasses.dataclass
+class SplunkProviderAuthConfig:
+    username: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "Splunk Username",
+        }
+    )
+    password: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "Splunk Password",
+            "sensitive": True,
+        }
+    )
```

Comment:
Isn't there a better way to do authentication with Splunk? (e.g. an API key or something)

Reply:
@talboren Yup we have API Keys, should we switch to that?